gamestonk_terminal/cryptocurrency/discovery/coinpaprika_view.py

```diff
@@ -0,0 +1,284 @@
+    )
+
+    try:
+
+        if other_args:
+            if not other_args[0][0] == "-":
+                other_args.insert(0, "-q")
+
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+
+        category = ns_parser.category
+        if category.lower() == "all":
+            category = "currencies,exchanges,icos,people,tags"
+
+        df = paprika.search(q=ns_parser.query, c=category)
+
+        if df.empty:
+            print(
+                f"No results for search query '{ns_parser.query}' in category '{ns_parser.category}'"
+            )
+            print("")
+            return
+
+        df = df.sort_values(by=ns_parser.sortby, ascending=ns_parser.descend)
+
+        print(
+            tabulate(
+                df.head(ns_parser.top),
+                headers=df.columns,
+                floatfmt=".1f",
+                showindex=False,
+                tablefmt="fancy_grid",
+            )
+        )
+        print("")
+
+    except Exception as e:
+        print(e, "\n")
+
+
+def coins(other_args: List[str]):
+    """Shows list of all available coins on CoinPaprika
+
+    Parameters
+    ----------
+    other_args: List[str]
+        Arguments to pass to argparse
+    """
+    parser = argparse.ArgumentParser(
+        add_help=False,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        prog="coins",
+        description="""Shows list of all available coins on CoinPaprika.
+        You can display top N number of coins with --top N flag,
+        You can search by starting letters with -l/--letter flag like `coins -l M`
+        And you can also specify by which column you are searching for coin with --key
+        Displays columns like:
+            rank, id, name, type""",
+    )
+    parser.add_argument(
+        "-s",
+        "--skip",
+        default=0,
+        dest="skip",
+        help="Skip n of records",
+        type=check_positive,
+    )
+    parser.add_argument(
+        "-t",
+        "--top",
+        default=30,
+        dest="top",
+        help="Limit of records",
+        type=check_positive,
+    )
+    parser.add_argument("-l", "--letter", dest="letter", help="First letters", type=str)
+    parser.add_argument(
+        "-k",
+        "--key",
+        dest="key",
+        help="Search in column symbol, name, id",
+        type=str,
+        choices=["id", "symbol", "name"],
+        default="symbol",
+    )
+
+    try:
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+
+        df = get_list_of_coins()
+
+        letter = ns_parser.letter
+        if letter and isinstance(letter, str):
+            df = df[
+                df[ns_parser.key].str.match(f"^({letter.lower()}|{letter.upper()})")
+            ]
+
+        try:
+            df = df[ns_parser.skip : ns_parser.skip + ns_parser.top]
+        except Exception:
+            df = get_list_of_coins()
```

Comment:
If it fails we should print an error, and not re-load the same dataframe. Otherwise, what I would do is not load against that dataframe, but instead copy the dataframe above so that that df is left untouched, if that makes sense